src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9337,69 +9352,154 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
+                || receiver.Type.IsTypeParameter() // PROTOTYPE need to confirm what we want for extension indexer access on type parameters
+                || analyzedArguments.HasDynamicArgument) // PROTOTYPE need to confirm what we want for dynamic access
+            {
+                extensionIndexerAccess = null;
+                return false;
+            }
+
+            foreach (var scope in new ExtensionScopes(this))
+            {
+                if (tryBindExtensionIndexerAccessInSingleBinder(
+                    receiver, analyzedArguments, scope.Binder, originalBinder: this, syntax, diagnostics, out extensionIndexerAccess))
+                {
+                    return true;
+                }
+            }
 
-                var isExpanded = resolutionResult.Result.Kind == MemberResolutionKind.ApplicableInExpandedForm;
-                var argsToParams = resolutionResult.Result.ArgsToParamsOpt;
+            extensionIndexerAccess = null;
+            return false;
 
-                ReportDiagnosticsIfObsolete(diagnostics, property, syntax, hasBaseReceiver: receiver != null && receiver.Kind == BoundKind.BaseReference);
+            static bool tryBindExtensionIndexerAccessInSingleBinder(
+                BoundExpression receiver, AnalyzedArguments analyzedArguments, Binder binder,
+                Binder originalBinder, SyntaxNode syntax, BindingDiagnosticBag diagnostics,
+                [NotNullWhen(true)] out BoundIndexerAccess? extensionIndexerAccess)
+            {
+                Debug.Assert(receiver.Type is not null);
 
-                // Make sure that the result of overload resolution is valid.
-                var gotError = MemberGroupFinalValidationAccessibilityChecks(receiver, property, syntax, diagnostics, invokedAsExtensionMethod: false);
+                var lookupResult = LookupResult.GetInstance();
+                CompoundUseSiteInfo<AssemblySymbol> lookupUseSiteInfo = originalBinder.GetNewCompoundUseSiteInfo(diagnostics);
 
-                receiver = ReplaceTypeOrValueReceiver(receiver, property.IsStatic, diagnostics);
+                binder.LookupImplicitExtensionMembersInSingleBinder(lookupResult, receiver.Type, WellKnownMemberNames.Indexer,
+                    arity: 0, basesBeingResolved: null, LookupOptions.Default, originalBinder, ref lookupUseSiteInfo);
 
-                this.CheckAndCoerceArguments<PropertySymbol>(resolutionResult, analyzedArguments, diagnostics, receiver, invokedAsExtensionMethod: false);
+                // PROTOTYPE test use-site diagnostics
+                diagnostics.Add(syntax, lookupUseSiteInfo);
 
-                if (!gotError && receiver != null && receiver.Kind == BoundKind.ThisReference && receiver.WasCompilerGenerated)
+                if (!lookupResult.IsMultiViable)
                 {
-                    gotError = IsRefOrOutThisParameterCaptured(syntax, diagnostics);
+                    lookupResult.Free();
```

Comment:
Looks like we don't free `lookupResult` outside of this `if`.

Reply:
Indeed. Thanks